### PR TITLE
fix(gv-confirm): add disabled attribute and handle slot gv-button dis…

### LINKED ISSUE #233: [gv-confirm] if gv-button is disabled, gv-confirm should not popup
I have the following: 
```
<gv-confirm message="Are you sure ?" okLabel="OK" cancelLabel="CANCEL">
   <gv-button disabled="true" danger>Action<gv-button>
</gv-confirm>
```

I would expect confirm not to popup when clicking on button as it is disabled.

Other solution would be to put a disabled attribute on gv-confirm.

### PR DESCRIPTION
**Pull request by loriepisicchio**
…abled

Closes #233

### CODE REVIEW

File: src/molecules/gv-confirm.js [hunk 1]
```diff
@@ -151,6 +160,31 @@ export class GvConfirm extends GvPopover {
...
+
+  _calculateIsDisabled () {
+    const slot = this.shadowRoot.querySelector('slot');
+    const childNodes = slot.assignedNodes({ flatten: true });
+    let isSlotButtonDisabled = false;
+    childNodes.forEach((node) => {
+      if (node.nodeType === Node.ELEMENT_NODE) {
+        isSlotButtonDisabled = node.getAttribute('disabled');
+      }
+    });
```
Comment (gaetanmaisse) [inline src/molecules/gv-confirm.js:186]: With this `forEach` we get the activation state of the last sub-node, I don't know if it is the intended behavior.
Here is a suggestion with `filter` + `some` to take into account all the sub node.
```suggestion
    const isSlotButtonDisabled = childNodes
      .filter((node) => node.nodeType === Node.ELEMENT_NODE)
      .some((node) => Boolean(node.getAttribute('disabled')) === true);
```

I also added a `Boolean(...)` cast as `node.getAttribute('disabled')` is returning a string, like `"true"`.

File: src/molecules/gv-confirm.js [hunk 2]
```diff
@@ -151,6 +160,31 @@ export class GvConfirm extends GvPopover {
...
+      this._reflectDisabledOnSlot();
+    }
+  }
+
+  _reflectDisabledOnSlot () {
+    const slot = this.shadowRoot.querySelector('slot');
+    const childNodes = slot.assignedNodes({ flatten: true });
+    childNodes.forEach((node) => {
+      node.disabled = true;
+      return node.disabled;
```
Comment (gaetanmaisse) [inline src/molecules/gv-confirm.js:174]: I think we can remove this `return`

**Review by gaetanmaisse — APPROVED** [review]
LGTM, tested locally it's working fine with the examples provided in the stories.

File: src/molecules/gv-confirm.js (hunk 1 above)
Comment (loriepisicchio) [reply src/molecules/gv-confirm.js]: :+1:

**Review by gaetanmaisse — APPROVED** [review]

**Pull request merged**